### PR TITLE
RDRP 778

### PR DESCRIPTION
**Pull request by cbrowning04**
## Pull Request submission

This PR aims to solve RDRP-778. This includes re-writing write_snapshot_to_feather so that there is more rationale for the functions actions. This function has also been renamed to 'df_to_feather'.

Tests are included within this PR.

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes #

#### Code

- [ ] **Code runs** The code runs on my machine and/or CDSW
- [ ] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [ ] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [ ] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `environment.yaml`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

File: src/staging/staging_main.py [hunk 1]
```diff
@@ -143,14 +143,23 @@ def run_staging(
...
                 feather_path,
-                snapshot_name,
+                feather_fname,
                 full_responses,
-                write_feather,
-                secondary_snapshot_name,
+                write_feather
+
+            )
+            helpers.df_to_feather(
```
Comment (JenCheshire) [inline src/staging/staging_main.py]: If secondary_full_responses = None, the pipeline fails when it tries to write to feather - can that be skipped when there's no secondary_full_responses

File: src/staging/staging_main.py [hunk 2]
```diff
@@ -143,14 +143,24 @@ def run_staging(
...
+
             )
+            if secondary_full_responses is not None:
+                helpers.df_to_feather(
+                    feather_path,
+                    s_feather_fname,
+                    secondary_full_responses,
+                    write_feather
+
+                )
```
Comment (JenCheshire) [inline src/staging/staging_main.py:164]: When I run it with the secondary_full_responses it is erroring at src.staging.spp_snapshot_processing - full_responses - INFO:Starting Data Transmutation... with KeyError: 'Only a column name can be used for the key in a dtype mappings argument.'

Comment (cbrowning04) [reply src/staging/staging_main.py:164]: can you post the full error?

Comment (JenCheshire) [reply src/staging/staging_main.py:164]: I have fixed this by adding the instance col to the secondary snapshot (fix already in place for DevTest)

**Review by zorge69 — COMMENTED** [review]
Automated linting by flake8 give the following errors:
line too long (94 > 88 characters)
line too long (118 > 88 characters), etc.

The flake8 VS Code plugin can be installed from https://code.visualstudio.com/docs/python/linting

The issue is, it does not work with Python 3.6, only 3.7 and higher. 

My workaround: Created conda envrironment "resdev" with Python 3.6. Also, used a default conda environment "base" with Python 3.8. Installed flake8 plugin for Python 3.8. Switched to base, checked the "PROBLEMS" tab. Then, to run the program, switch back to "resdev" environment.

**Review by zorge69 — APPROVED** [review]
Fixed style issues with flake8

**Pull request merged**